### PR TITLE
Fixing broken PR

### PR DESCRIPTION
**Pull request by Eb-Zeero**
This change is [Reviewable](https://reviewable.io/reviews/saltastroops/web-manager-2021/11)

### CODE REVIEW

**Review by hettlage — CHANGES_REQUESTED** [review]
> Reviewed 2 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/web-manager-2021/11)* status: 2 of 3 files reviewed, 26 unresolved discussions (waiting on @Eb-Zeero)

___
*[python/app/service/proposal.py, line 9 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFD7rxAiPbEX9jV2RR:-MYFD7ry42e74XE89WAL:b-dzcwqd) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L9)):*
> ```Python
> 
> 
> async def get_proposal_text(
> ```

get_proposal_text ---> get_text_content
___
*[python/app/service/proposal.py, line 20 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFHHVj5Bc-OGUFxaN1:-MYFHHVj5Bc-OGUFxaN2:bbicuy) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L20)):*
> ```Python
>     """
>     async with db.acquire() as conn:
>         async with conn.cursor() as cur:
> ```

Consider using a dict cursor (if available).

That's true for all the SQL queries.
___
*[python/app/service/proposal.py, line 31 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFAtGlB7OQRWJ5faHN:-MYFAtGlB7OQRWJ5faHO:b-tc9g0e) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L31)):*
> ```Python
>             rs = await cur.fetchone()
>             if rs:
>                 return dict(
> ```

There should be a model for this.
___
*[python/app/service/proposal.py, line 35 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFBEdn2sgNoo1X8oWk:-MYFBEdo9JXC5CUZlcqf:b-rf3g8m) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L35)):*
> ```Python
>                     abstract=rs[1],
>                     read_me=rs[2],
>                     night_summary=rs[3],
> ```

"night_summary" should read "nightlog_summary" or even "summary_for_nightlog".
___
*[python/app/service/proposal.py, line 37 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFHbg11ufoWQqkGg-s:-MYFHbg11ufoWQqkGg-t:b5m05bj) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L37)):*
> ```Python
>                     night_summary=rs[3],
>                 )
>     raise ValueError("Proposal text Could not be found.")
> ```

The error might include the proposal code.
___
*[python/app/service/proposal.py, line 40 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFHg3fEQXovCm5VEaI:-MYFHg3fEQXovCm5VEaJ:b-eemwu1) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L40)):*
> ```Python
> 
> 
> async def get_proposal_investigators(
> ```

get_proposal_investigators ---> get_investigators
___
*[python/app/service/proposal.py, line 45 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFH1Oo4Iwa6djkIhDO:-MYFH1Oo4Iwa6djkIhDP:bofah7d) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L45)):*
> ```Python
>     sql = """\
> SELECT pi.Investigator_Id, FirstName, Surname, Partner_Name, InstituteName_Name,
>         Department, Url, Leader_Id, Contact_Id FROM ProposalInvestigator AS pi
> ```

"FROM" should start on a new line. You might consider having one field per line.
___
*[python/app/service/proposal.py, line 59 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFGIBAAFaU4JBPL8Ya:-MYFGIBAAFaU4JBPL8Yb:bukensb) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L59)):*
> ```Python
>             rs = await cur.fetchall()
>             if rs:
>                 return [
> ```

Should be a model.

pc --> is_pc
pi --> is_pi
I would not combine the first name and last name

One might consider using a separate model for the affiliation.

"home_page" is somewhat ambiguous.
___
*[python/app/service/proposal.py, line 74 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFIRkSAG6fIg7ocy5W:-MYFIRkSAG6fIg7ocy5X:b-6iexmb) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L74)):*
> ```Python
> 
> 
> async def get_proposal_allocations(
> ```

get_proposal_allocations ---> get_time_allocations
___
*[python/app/service/proposal.py, line 100 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFJTRF7dHuvnmFFbR-:-MYFJTRF7dHuvnmFFbR0:bszxgfk) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L100)):*
> ```Python
>             rs = await cur.fetchall()
>             if rs:
>                 _alloc = {}
> ```

Should be a model (which needs discussion).
___
*[python/app/service/proposal.py, line 115 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFM36y7N2NWd-GvZpn:-MYFM36y7N2NWd-GvZpo:b-w49mq5) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L115)):*
> ```Python
> async def get_proposal_targets(proposal_code: str, db: connect) -> List[Dict[str, Any]]:
>     sql = """
> SELECT Target_Name, RaH, RaM, RaS, DecSign, DecD, DecM, DecS, Equinox, MinMag, MaxMag,
> ```

This query will miss non-sidereal targets, I think.

Also, it will return targets for all submissions.

Also, it is for Phase 1 only -- but the function name doesn't reflect this.
___
*[python/app/service/proposal.py, line 134 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFKH8N3-RLjFHX7s9C:-MYFKH8N3-RLjFHX7s9D:bszxgfk) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L134)):*
> ```Python
>             rs = await cur.fetchall()
>             if rs:
>                 return [
> ```

Should be a model (which needs discussion).
___
*[python/app/service/proposal.py, line 147 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFLcaP9UAyGe6aD48V:-MYFLcaP9UAyGe6aD48W:bwvtg4i) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L147)):*
> ```Python
>                         min_mag=r[9],
>                         max_mag=r[10],
>                         type=r[11],
> ```

As type is a Python keyword, maybe better target_type (and then target_subtype).
___
*[python/app/service/proposal.py, line 149 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFLr_i2nLqgd0P24PF:-MYFLr_i2nLqgd0P24PG:b3pbzia) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L149)):*
> ```Python
>                         type=r[11],
>                         sub_type=r[12],
>                         optional=r[13],
> ```

Maybe better is_optional
___
*[python/app/service/proposal.py, line 150 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFQSXoB3J0RySaKKMx:-MYFQSXoB3J0RySaKKMy:b-acebm4) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L150)):*
> ```Python
>                         sub_type=r[12],
>                         optional=r[13],
>                         nvisits=r[14],
> ```

Might just be visits (or n_visits).
___
*[python/app/service/proposal.py, line 152 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFLuXo6xCn4BJtA9Ua:-MYFLuXo6xCn4BJtA9Ub:b-u8a3o) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L152)):*
> ```Python
>                         nvisits=r[14],
>                         max_luner_phase=r[15],
>                         raking=[16],
> ```

raking ---> ranking
___
*[python/app/service/proposal.py, line 164 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFQYdV1nvTf-r99_nG:-MYFQYdV1nvTf-r99_nH:b-lise39) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L164)):*
> ```Python
> 
> 
> async def get_observed_targets(
> ```

get_observed_targets ---> get_block_visits
___
*[python/app/service/proposal.py, line 188 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFRj7jADV2Ju9zRQ-2:-MYFRj7jADV2Ju9zRQ-3:b-jwhjb1) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L188)):*
> ```Python
>             await cur.execute(sql, {"proposal_code": proposal_code})
>             rs = await cur.fetchall()
>             if rs:
> ```

The way this is handled is problematic if there is more than one pointing or observation.
___
*[python/app/service/proposal.py, line 189 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFRF1J6KSU5SVPRG_K:-MYFRF1J6KSU5SVPRG_L:btuubqx) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L189)):*
> ```Python
>             rs = await cur.fetchall()
>             if rs:
>                 return [
> ```

Should be a model.
___
*[python/app/service/proposal.py, line 198 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFVdD8A_hZx6IZ7-TB:-MYFVdD8A_hZx6IZ7-TC:bkam0v5) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L198)):*
> ```Python
>                         target_name=r[5],
>                         observation_date=r[6],
>                         block_visit_status=r[7],
> ```

block_visit_status ---> status
block_rejection_reason ---> rejection_reason
___
*[python/app/service/proposal.py, line 206 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFRvWe8P_JC9YKqGJ1:-MYFRvWe8P_JC9YKqGJ2:b-mlqtjh) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L206)):*
> ```Python
> 
> 
> async def get_proposal_requested_time(
> ```

get_proposal_requested_time ---> get_requested_time
___
*[python/app/service/proposal.py, line 228 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFS3lI1q4_wZWLyZiR:-MYFS3lI1q4_wZWLyZiS:bluqhw5) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L228)):*
> ```Python
>     async with db.acquire() as conn:
>         async with conn.cursor() as cur:
>             await cur.execute(requested_time_sql, {"proposal_code": proposal_code})
> ```

You should execute these queries in parallel.
___
*[python/app/service/proposal.py, line 231 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFSvsODrLpBs4is7uF:-MYFSvsODrLpBs4is7uG:b-iao4e4) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L231)):*
> ```Python
>             rrs = await cur.fetchall()
>             for r in rrs:
>                 _sem = f"{r[3]}-{r[4]}"
> ```

_sem might need discussion.
___
*[python/app/service/proposal.py, line 250 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFVM_O2pTAfzGqn7bY:-MYFVM_O2pTAfzGqn7bZ:b-fjd9nr) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L250)):*
> ```Python
>                     )
>             _req = []
>             for r in requested_time:
> ```

This looks odd?
___
*[python/app/service/proposal.py, line 260 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFVwJJ1KrnqZp3oGPn:-MYFVwJJ1KrnqZp3oGPo:bd4vg7o) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L260)):*
> ```Python
> ) -> Dict[str, Any]:
>     sql = """
> SELECT SUM(ObsTime), Priority FROM BlockVisit AS bv
> ```

FROM should start on a new line.
___
*[python/app/service/proposal.py, line 282 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFW8KA-KXcD0S9l5Y4:-MYFW8KA-KXcD0S9l5Y5:btuubqx) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L282)):*
> ```Python
>             )
>             rs = await cur.fetchall()
>             observed_time = dict(
> ```

Should be a model.

**Review by Eb-Zeero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/saltastroops/web-manager-2021/11)* status: 2 of 3 files reviewed, 26 unresolved discussions (waiting on @Eb-Zeero and @hettlage)

___
*[python/app/service/proposal.py, line 9 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFD7rxAiPbEX9jV2RR:-MYKyBtjEJFuUfJb6277:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L9)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

get_proposal_text ---> get_text_content
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 31 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFAtGlB7OQRWJ5faHN:-MYL7-MT-fIxcKYH2CcY:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L31)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

There should be a model for this.
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 35 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFBEdn2sgNoo1X8oWk:-MYL7BJq0zvSovNv8OYM:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L35)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

"night_summary" should read "nightlog_summary" or even "summary_for_nightlog".
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 37 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFHbg11ufoWQqkGg-s:-MYd_jUcEpEIhvAgvAL8:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L37)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

The error might include the proposal code.
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 40 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFHg3fEQXovCm5VEaI:-MYL7aNj7H9uU3_CvTpQ:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L40)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

get_proposal_investigators ---> get_investigators
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 45 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFH1Oo4Iwa6djkIhDO:-MYL7aoX5pO0d5FlUTO3:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L45)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

"FROM" should start on a new line. You might consider having one field per line.
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 59 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFGIBAAFaU4JBPL8Ya:-MYL7cnGDfJV8UBRxPCE:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L59)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Should be a model.

pc --> is_pc
pi --> is_pi
I would not combine the first name and last name

One might consider using a separate model for the affiliation.

"home_page" is somewhat ambiguous.
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 74 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFIRkSAG6fIg7ocy5W:-MYL7jhT0ec9DCSl9seh:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L74)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

get_proposal_allocations ---> get_time_allocations
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 100 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFJTRF7dHuvnmFFbR-:-MYdQTzD1DUxgsuvzXje:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L100)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Should be a model (which needs discussion).
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 115 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFM36y7N2NWd-GvZpn:-MYFv3cy9ze9h-nHOnrD:b3y98z) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L115)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This query will miss non-sidereal targets, I think.

Also, it will return targets for all submissions.

Also, it is for Phase 1 only -- but the function name doesn't reflect this.
</blockquote></details>

left join horizonTable and movingTarget
___
*[python/app/service/proposal.py, line 134 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFKH8N3-RLjFHX7s9C:-MYdYqiQ3dF-UhyuGPhY:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L134)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Should be a model (which needs discussion).
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 147 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFLcaP9UAyGe6aD48V:-MYOUwKE44tpRd2T8hup:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L147)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

As type is a Python keyword, maybe better target_type (and then target_subtype).
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 149 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFLr_i2nLqgd0P24PF:-MYOUwl_6vi-FlmFZ3h_:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L149)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Maybe better is_optional
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 150 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFQSXoB3J0RySaKKMx:-MYOUzTvC4RBkofr3jRp:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L150)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Might just be visits (or n_visits).
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 152 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFLuXo6xCn4BJtA9Ua:-MYOV6wg05QO-FXms90M:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L152)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

raking ---> ranking
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 164 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFQYdV1nvTf-r99_nG:-MYOVE7K3Tx-QeIOobzu:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L164)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

get_observed_targets ---> get_block_visits
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 189 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFRF1J6KSU5SVPRG_K:-MYOZieK6A3974ZUKd6w:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L189)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Should be a model.
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 198 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFVdD8A_hZx6IZ7-TB:-MYOVfVy92BsfcNhbq3c:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L198)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

block_visit_status ---> status
block_rejection_reason ---> rejection_reason
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 206 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFRvWe8P_JC9YKqGJ1:-MYOZtDvD0IjHtvEJtD_:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L206)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

get_proposal_requested_time ---> get_requested_time
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 228 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFS3lI1q4_wZWLyZiR:-MYG2hkY9Yu5Vm5cj5X_:bsnn6uh) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L228)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

You should execute these queries in parallel.
</blockquote></details>

+use the asyncio gather  method+
the use of Gather will result in having multiples cursors which is not recomended or not a good practice
___
*[python/app/service/proposal.py, line 231 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFSvsODrLpBs4is7uF:-MYd_X1N4itYR8zYnrE7:b-qssebl) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L231)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

_sem might need discussion.
</blockquote></details>

fixed.
___
*[python/app/service/proposal.py, line 250 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFVM_O2pTAfzGqn7bY:-MYd_bGv1oKhEQt8A1Fo:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L250)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This looks odd?
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 260 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFVwJJ1KrnqZp3oGPn:-MYd-YTFEjehOwuD0WrX:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L260)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

FROM should start on a new line.
</blockquote></details>

Done.
___
*[python/app/service/proposal.py, line 282 at r1](https://reviewable.io/reviews/saltastroops/web-manager-2021/11#-MYFW8KA-KXcD0S9l5Y4:-MYd_fVJ8sTHMb2bQIEX:b-896fix) ([raw file](https://github.com/saltastroops/web-manager-2021/blob/88c318e596b1a447507cb7cfbc773f26aaf3767c/python/app/service/proposal.py#L282)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Should be a model.
</blockquote></details>

Done.

File: python/app/service/proposal.py [hunk 1]
```diff
@@ -190,8 +192,9 @@ async def get_block_visits(proposal_code: str, db: connect) -> List[BlockVisit]:
     LEFT JOIN BlockRejectedReason AS brr
         ON brr.BlockRejectedReason_Id = bv.BlockRejectedReason_Id
 WHERE Proposal_Code = %(proposal_code)s
-GROUP BY bv.Block_Id 
-    """  # TODO: can have more than one pointing and observation per block visit
+    """
```
Comment (hettlage) [inline python/app/service/proposal.py:195]: Don't you have to group by BlockVisit_Id?

File: python/app/service/proposal.py [hunk 2]
```diff
@@ -331,3 +315,32 @@ async def get_observed_time(
             for r in rs:
                 observed_time[f"priority_{r[1]}"] = r[0]
             return observed_time
+
+
+def get_proposal(proposal_code, semester: Semester, db: connect):
+    text_content = get_text_content(proposal_code, semester, db)
```
[the remark this answers is not in the document]
Comment (hettlage) [reply python/app/service/proposal.py:321]: And you are missing await. :-)

File: python/app/service/proposal.py [hunk 3]
```diff
@@ -284,7 +266,9 @@ async def get_requested_time(
             await cur.execute(distribution_time_sql, {"proposal_code": proposal_code})
             drs = await cur.fetchall()
             for r in drs:
-                if not (r[2] == "OTH"):  # TODO: raise error if oth allocated time
+                if r[2] == "OTH" and r[4] > 0:
+                    raise ValueError("The partner other should not have share time.")
+                if not (r[2] == "OTH"):
                     _sem = f"{r[0]}-{r[1]}"
                     requested_time[_sem].distribution.append(
                         dict(
```
Comment (hettlage) [inline python/app/service/proposal.py:274]: This should be a type rather than a dict.

**Review by hettlage — CHANGES_REQUESTED** [review]

File: python/app/service/proposal.py [hunk 4]
```diff
@@ -284,7 +266,9 @@ async def get_requested_time(
             await cur.execute(distribution_time_sql, {"proposal_code": proposal_code})
             drs = await cur.fetchall()
             for r in drs:
-                if not (r[2] == "OTH"):  # TODO: raise error if oth allocated time
+                if r[2] == "OTH" and r[4] > 0:
+                    raise ValueError("The partner other should not have share time.")
```
[the remark this answers is not in the document]
Comment (Eb-Zeero) [reply python/app/service/proposal.py:270]: removed

File: python/app/service/proposal.py (hunk 1 above)
Comment (Eb-Zeero) [reply python/app/service/proposal.py:195]: yes i do and I did some how keeps on reverting back

File: python/app/service/proposal.py (hunk 3 above)
Comment (Eb-Zeero) [reply python/app/service/proposal.py:274]: done

**Pull request closed without merging**